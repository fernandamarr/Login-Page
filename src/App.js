import React from 'react';
import { BrowserRouter as  Router, Route, Switch } from "react-router-dom";
import './components/style.css';
import Login from './components/Login';
import Reset from './components/Reset';
import Signup from './components/Signup';

function App() {
  return (
    <Router>
      <div className="App">
        <Switch>            
          <Route exact path="/" component={Login} />
          <Route path="/reset" component={Reset} />
          <Route path="/signup" component={Signup} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
