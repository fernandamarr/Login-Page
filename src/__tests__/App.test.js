import React from 'react';
import App from '../App';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer'

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('handles validation', () => {
	expect(formIsValid).toBe(false);
	instance.handleValidation();
	expect(formIsValid)toBe(true);
	expect(tree.toJSON()).toMatchSnapshot()

})