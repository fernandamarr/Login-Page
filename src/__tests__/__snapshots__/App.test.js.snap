// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app button 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="container"
    >
      <form
        name="form"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="email"
            value=""
          />
          <div
            className="errorMsg"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <div
            className="errorMsg"
          />
        </div>
        <button
          className="btn btn-primary"
          type="submit"
        >
          Login
        </button>
        <div
          className="links"
        >
          <a
            className="links"
            href="/reset"
            onClick={[Function]}
          >
            Forgot my password
          </a>
          <span
            className="divider"
          >
             | 
          </span>
          <a
            className="links"
            href="/signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </div>
      </form>
    </div>
  </header>
</div>
`;
